testcases/parser/nodes.go

```diff
@@ -78,6 +78,25 @@ func (c *CaseLiteral) AsAggregateArgumentString() string {
 	return c.Value.ValueString() + "::" + c.Type.String()
 }
 
+// change the type of the literal to the given type
+func (c *CaseLiteral) fixTypeInTheLiteral() error {
```

Comment:
It's not clear to me looking at this function what fixing does.  The same goes for the error message.  It looks like the parser is treating the type declaration as a cast operation.